scripts/src/ledger.ts

```diff
@@ -1,12 +1,24 @@
 // Adapted from @solana/wallet-adapter-ledger, with some modifications to support Node.js
+import fs from "node:fs";
+
+import { Wallet } from "@coral-xyz/anchor";
 import Transport, {
   StatusCodes,
   TransportStatusError,
 } from "@ledgerhq/hw-transport";
 import TransportNodeHid from "@ledgerhq/hw-transport-node-hid";
 import { getDerivationPath } from "@solana/wallet-adapter-ledger";
 import type { Transaction } from "@solana/web3.js";
-import { VersionedTransaction, PublicKey } from "@solana/web3.js";
+import { Keypair, VersionedTransaction, PublicKey } from "@solana/web3.js";
+
+export const parseSignerSource = async (source: string) =>
+  source.startsWith("usb://ledger")
```

Comment:
q: should we do a case-insensitive check via a RegExp or `String.prototype..toLowerCase()` first, or is it safe to rely on the string already being lowercased?